platform/core/src/poly_out.rs

```diff
@@ -0,0 +1,112 @@
+use std::fs::DirBuilder;
+use std::path::Path;
+use uuid::Uuid;
+
+type ResourceId = String;
+type FsId = String;
+
+struct Metadata {}
+struct Content {}
+
+trait PolyOutTrait {
+    fn metadata(resource_id: ResourceId) -> Result<Metadata, String>;
+    fn read(resource_id: ResourceId) -> Result<Content, String>;
+    fn remove(resource_id: ResourceId) -> Result<(), String>;
```

Comment:
Would this remove directories, or just files? If it's the former, recursively, or not? Lots of design questions - I'm OK with having a single call that does all that, but that'd need some doc strings then.

Reply:
Both. If you give it a url, it will remove the contents, file or directory. 

> Would this remove directories, or just files? If it's the former, recursively, or not?

If you remove a directory, its contents will also be removed.